ytmusicapi/auth/oauth.py

```diff
@@ -22,81 +22,186 @@ def is_oauth(headers: CaseInsensitiveDict) -> bool:
     return all(key in headers for key in oauth_structure)
 
 
-def is_custom_oauth(headers: CaseInsensitiveDict) -> bool:
-    """Checks whether the headers contain a Bearer token, indicating a custom OAuth implementation."""
-    return "authorization" in headers and headers["authorization"].startswith("Bearer ")
-
-
-class YTMusicOAuth:
-    """OAuth implementation for YouTube Music based on YouTube TV"""
-
-    def __init__(self, session: requests.Session, proxies: Dict = None):
-        self._session = session
+class OAuthToken:
```

Comment:
since we have multiple classes now, we can easily split this across 3 files in an `oauth` folder to keep it modular